### ISSUE TITLE
FTBFS at Debian: Creation method of abstract class cannot be public.

### ISSUE THREAD

**Issue by JoergFF** [Schneegans/Gnome-Pie#177]
Hello,

on a Debian Sid / Bullseye build[1] I get:

```
/build/gnome-pie-0.7.2/src/actions/action.vala:72.5-72.17: error: Creation method of abstract class cannot be public.
    public Action(string name, string icon, bool is_quickaction) {

```

Please can you check it?
Many thanks

CU
Jörg

[1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=948129

**Comment by JoergFF** [comment]
Hi,

the build log ist attached.

CU
Jörg
[gnome-pie_0.7.2-2_amd64.build.txt](https://github.com/Schneegans/Gnome-Pie/files/4023286/gnome-pie_0.7.2-2_amd64.build.txt)

**Comment by corebots** [comment]
Hi,

i'm on Fedora 32 beta, after installing dependencies, running `./make.sh` gives following error:

`/Gnome-Pie/src/actions/action.vala:72.5-72.17: error: Creation method of abstract class cannot be public.
    public Action(string name, string icon, bool is_quickaction) {
    ^^^^^^^^^^^^^
Compilation failed: 1 error(s), 0 warning(s)
make[2]: *** [src/CMakeFiles/gnome-pie.dir/build.make:142: src/actionGroups/actionGroup.c] Error 1
make[1]: *** [CMakeFiles/Makefile2:138: src/CMakeFiles/gnome-pie.dir/all] Error 2
make: *** [Makefile:147: all] Error 2
`

**Comment by Skrrytch** [comment]
The same here with Fedora 32 (final). 

It may not be the final solution but removing "public" from line 72 solved the problem:

    Action(string name, string icon, bool is_quickaction) {
        GLib.Object(name : name, icon : icon, is_quickaction : is_quickaction);
    }

**Comment by kpcyrd** [comment]
Same issue on Arch Linux, would be neat if this could be fixed in a new release. :)
```
==> Starting build()...
bg/ 101 translated messages, 55 untranslated messages.
de/ 156 translated messages.
es/ 156 translated messages.
fr/ 153 translated messages, 3 untranslated messages.
it/ 47 translated messages, 19 fuzzy translations, 90 untranslated messages.
ko/ 99 translated messages, 1 fuzzy translation, 56 untranslated messages.
lt/ 156 translated messages.
nl_NL/ 153 translated messages.
pl/ 63 translated messages, 93 untranslated messages.
pt_BR/ 101 translated messages, 55 untranslated messages.
ru/ 155 translated messages, 1 untranslated message.
zh_CN/ 101 translated messages, 55 untranslated messages.
-- The C compiler identification is GNU 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Deprecation Warning at CMakeLists.txt:7 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Found Vala: /usr/bin/valac  
-- checking for a minimum Vala version of 0.22.0
--   found Vala, version 0.52.4
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.7.3") 
-- Checking for module 'gtk+-3.0'
--   Found gtk+-3.0, version 3.24.30
-- Checking for module 'appindicator3-0.1'
--   Package 'appindicator3-0.1', required by 'virtual:world', not found
-- Checking for module 'libwnck-3.0'
--   Found libwnck-3.0, version 40.0
-- Checking for module 'libbamf3'
--   Package 'libbamf3', required by 'virtual:world', not found
-- Checking for module 'gio-unix-2.0'
--   Found gio-unix-2.0, version 2.68.3
-- Checking for module 'cairo'
--   Found cairo, version 1.17.4
-- Checking for module 'gee-0.8'
--   Found gee-0.8, version 0.20.4
-- Checking for module 'x11'
--   Found x11, version 1.7.2
-- Checking for module 'libxml-2.0'
--   Found libxml-2.0, version 2.9.10
-- Checking for module 'xtst'
--   Found xtst, version 1.2.3
-- Checking for module 'libgnome-menu-3.0'
--   Found libgnome-menu-3.0, version 3.36.0

Summary:
  EXECUTABLE_OUTPUT_PATH = /build/gnome-pie/src/Gnome-Pie-0.7.2
  VALA_EXECUTABLE = /usr/bin/valac
Libs:
  GTK3_LIBRARIES = gtk-3;gdk-3;z;pangocairo-1.0;pango-1.0;harfbuzz;atk-1.0;cairo-gobject;cairo;gdk_pixbuf-2.0;gio-2.0;gobject-2.0;glib-2.0
  CAIRO_LIBRARIES = cairo
  GEE_LIBRARIES = gee-0.8;gobject-2.0;glib-2.0
  INDICATOR_LIBRARIES = 
  INDICATOR3_LIBRARIES = 
  XML_LIBRARIES = xml2
  XTST_LIBRARIES = Xtst
  GMENU_LIBRARIES = 
  GMENU3_LIBRARIES = gnome-menu-3;gio-2.0;gobject-2.0;glib-2.0
  X11_LIBRARIES = X11
  WNCK_LIBRARIES = wnck-3;gtk-3;gdk-3;z;pangocairo-1.0;pango-1.0;harfbuzz;atk-1.0;cairo-gobject;cairo;gdk_pixbuf-2.0;gio-2.0;gobject-2.0;glib-2.0
  ARCHIVE_LIBRARIES = /usr/lib/libarchive.so

-- Configuring done
-- Generating done
-- Build files have been written to: /build/gnome-pie/src/Gnome-Pie-0.7.2/build
[  1%] Generating actionGroups/actionGroup.c, actionGroups/bookmarkGroup.c, actionGroups/clipboardGroup.c, actionGroups/devicesGroup.c, actionGroups/groupRegistry.c, actionGroups/menuGroup.c, actionGroups/sessionGroup.c, actionGroups/windowListGroup.c, actions/action.c, actions/actionRegistry.c, actions/appAction.c, actions/keyAction.c, actions/pieAction.c, actions/sigAction.c, actions/uriAction.c, daemon.c, gui/aboutWindow.c, gui/commandComboList.c, gui/iconSelectWindow.c, gui/indicator.c, gui/newSliceWindow.c, gui/newsWindow.c, gui/pieComboList.c, gui/pieList.c, gui/pieOptionsWindow.c, gui/piePreview.c, gui/piePreviewAddSign.c, gui/piePreviewCenter.c, gui/piePreviewDeleteSign.c, gui/piePreviewRenderer.c, gui/piePreviewSliceRenderer.c, gui/preferencesWindow.c, gui/sliceTypeList.c, gui/themeList.c, gui/tipViewer.c, gui/triggerSelectButton.c, images/icon.c, images/image.c, images/renderedText.c, images/themedIcon.c, pies/defaultConfig.c, pies/load.c, pies/pie.c, pies/pieManager.c, pies/save.c, renderers/centerRenderer.c, renderers/pieRenderer.c, renderers/pieWindow.c, renderers/sliceRenderer.c, themes/centerLayer.c, themes/sliceLayer.c, themes/theme.c, themes/themeImporter.c, utilities/animatedValue.c, utilities/archiveReader.c, utilities/archiveWriter.c, utilities/bindingManager.c, utilities/color.c, utilities/config.c, utilities/focusGrabber.c, utilities/key.c, utilities/logger.c, utilities/paths.c, utilities/trigger.c
/build/gnome-pie/src/Gnome-Pie-0.7.2/src/actions/action.vala:72.5-72.17: error: Creation method of abstract class cannot be public.
    public Action(string name, string icon, bool is_quickaction) {
    ^^^^^^^^^^^^^
Command-line option `--thread` is deprecated and will be ignored
Compilation failed: 1 error(s), 0 warning(s)
make[2]: *** [src/CMakeFiles/gnome-pie.dir/build.make:137: src/actionGroups/actionGroup.c] Error 1
make[1]: *** [CMakeFiles/Makefile2:124: src/CMakeFiles/gnome-pie.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
==> ERROR: A failure occurred in build().
```

**Comment by Schneegans** [comment]
This had actually been fixed about a year ago in `develop`. I just merged it to master and tagged it as `v0.7.3`!
